### PR TITLE
CertificateTroubleshoot: add openssl client support

### PR DESCRIPTION
**Pull request by glicht**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
* Add option to use openssl client for endpoint detection
* Limit the number of custom certificates returned
* Add debug logging
* Add small pack readme
 
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py [hunk 1]
```diff
@@ -153,39 +154,53 @@ def docker_container_details() -> dict:
...
+        certificates = "" if not container_ca_file.is_file() else container_ca_file.read_text()
+        custom_certs = parse_all_certificates(certificates)
+        demisto.debug(f'custom certs len: {len(custom_certs)}')
     return {
         "ShellVariables": {
             "SSL_CERT_FILE": os.environ.get('SSL_CERT_FILE'),
             "CERT_FILE": os.environ.get('REQUESTS_CA_BUNDLE'),
         },
-        "CustomCertificateAuthorities": parse_all_certificates(certificates),
+        "CustomCertificateAuthorities": custom_certs[:5],
```
Comment (Itay4) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py:174]: shouldnt this be configurable ?

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py [hunk 2]
```diff
@@ -153,39 +154,53 @@ def docker_container_details() -> dict:
...
+        demisto.debug(f'openssl output: {openssl_res}')
+        return '\n'.join(re.findall(r'^-----BEGIN CERT.*?^-----END CERTIFICATE-----', openssl_res, re.DOTALL | re.MULTILINE))
+    else:
+        hostname = endpoint
+        conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+        sock = context.wrap_socket(conn, server_hostname=hostname)
+        sock.settimeout(60)  # make sure we don't get stuck
+        sock.connect((hostname, int(port)))
+        return ssl.DER_cert_to_PEM_cert(sock.getpeercert(True))  # type: ignore
```
Comment (Itay4) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py:203]: can we add the error code for that mypy ignore?

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py (hunk 1 above)
Comment (glicht) [reply Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py:174]: I think it's ok. It's for safety not to overloaded the response. I don't want to over engineer things.

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py (hunk 2 above)
Comment (glicht) [reply Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.py]: fixed

**Review by Itay4 — APPROVED** [review]

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.yml [hunk 3]
```diff
@@ -5,7 +5,17 @@ args:
   name: endpoint
   required: true
   secret: false
-- default: true
+- default: false
+  description: Operation mode. Determines how the endpoint is inspected. Either python or openssl.
```
Comment (amshamah419) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/CertificatesTroubleshoot.yml:9]:
```suggestion
  description: Operation mode. Determines how the endpoint is inspected. Options are either python or openssl.
```

File: Packs/Troubleshoot/ReleaseNotes/1_1_1.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+##### CertificatesTroubleshoot
+- Added an option to detect endpoint certificates using the openssl client.
+- Improved debug logging.
+- Updated the Docker image to: *demisto/ssl-analyze:1.0.0.14890*.
```
Comment (amshamah419) [inline Packs/Troubleshoot/ReleaseNotes/1_1_1.md:6]:
```suggestion
#### Scripts
##### CertificatesTroubleshoot
- Added an option to detect endpoint certificates using the OpenSSL client.
- Improved debug logging.
- Updated the Docker image to: *demisto/ssl-analyze:1.0.0.14890*.
```

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md [hunk 5]
```diff
@@ -6,6 +6,13 @@ This automation exports all custom certificate-related information from the Pyth
 
 After following the [tutorial](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/docker/configure-python-docker-integrations-to-trust-custom-certificates) to update your custom certificate in `Cortex XSOAR Server`/ `Cortex XSOAR Engine`, validate the configuration applied using this script.
 
+The Script supports two modes of operation:
```
Comment (amshamah419) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md:9]:
```suggestion
The script supports two modes of operation:
```

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md [hunk 6]
```diff
@@ -6,6 +6,13 @@ This automation exports all custom certificate-related information from the Pyth
...
+
+  1. **python**: will use the python built-in SSL library to detect the endpoint's certificates.
```
Comment (amshamah419) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md:11]:
```suggestion
  1. **python**: Uses the Python built-in SSL library to detect the endpoint's certificates.
```

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md [hunk 7]
```diff
@@ -6,6 +6,13 @@ This automation exports all custom certificate-related information from the Pyth
...
+  2. **openssl**: will use the openssl client to detect the endpoint's certificates. Use this mode if the `python` mode fails for some reason.
```
Comment (amshamah419) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md:12]:
```suggestion
  2. **openssl**: Uses the OpenSSL client to detect the endpoint's certificates. Use this mode if the `python` mode fails for some reason.
```

File: Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md [hunk 8]
```diff
@@ -23,6 +30,7 @@ After following the [tutorial](https://docs.paloaltonetworks.com/cortex/cortex-x
 | --- | --- |
 | endpoint | The endpoint identifier IP address or URL:Port. If the port is not included, 443 will be used by default. |
 | port | The endpoint port. Default is 443. |
+| mode | Operation mode. Determines how the endpoint is inspected. Either using python built-in SSL or openssl client. |
```
Comment (amshamah419) [inline Packs/Troubleshoot/Scripts/CertificatesTroubleshoot/README.md:33]:
```suggestion
| mode | Operation mode. Determines how the endpoint is inspected. Either using Python built-in SSL or OpenSSL client. |
```

**Review by amshamah419 — COMMENTED** [review]
Looks good, just commit the changes.

**Pull request merged**